File: src/programmers/internal/json_schema_number.ts
import { OpenApi } from "@samchon/openapi";

import { MetadataAtomic } from "../../schemas/metadata/MetadataAtomic";

import { json_schema_plugin } from "./json_schema_plugin";

export const json_schema_number = (
  atomic: MetadataAtomic,
): Array<OpenApi.IJsonSchema.IInteger | OpenApi.IJsonSchema.INumber> =>
  json_schema_plugin({
    schema: {
      type: "number",
    } satisfies OpenApi.IJsonSchema.INumber,
    tags: atomic.tags,
  });

File: packages/typia/native/core/programmers/iterate/json_schema_number.go
package iterate

import nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"

func json_schema_number(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
	return json_schema_plugin(struct {
		schema JsonSchema
		tags   [][]nativemetadata.IMetadataTypeTag
	}{
		schema: JsonSchema{"type": "number"},
		tags:   atomic.Tags,
	})
}

func Json_schema_number_export(atomic *nativemetadata.MetadataAtomic) []JsonSchema {
	return json_schema_number(atomic)
}
